File: BhaktiChatAndroidNative/app/src/main/java/com/bhaktichat/app/ui/screens/explore/ExploreScreen.kt
package com.bhaktichat.app.ui.screens.explore

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.MenuBook
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material.icons.filled.CalendarMonth
import androidx.compose.material.icons.filled.Celebration
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.filled.GridView
import androidx.compose.material.icons.filled.MusicNote
import androidx.compose.material.icons.filled.Stars
import androidx.compose.material.icons.filled.WbSunny
import androidx.compose.material.icons.outlined.Person
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.bhaktichat.app.R
import com.bhaktichat.app.ui.components.shell.AppTopBar
import com.bhaktichat.app.ui.components.shell.BhaktiBottomNavBarDefaults

/** Warm saffron/cream palette for the new 4-tab IA screens (design_handoff_bhaktichat_ia). */
object ExplorePalette {
    val Card = Color(0xFFFFFFFF)
    val CardBorder = Color(0x1A784028)       // rgba(120,80,40,0.10)
    val TextPrimary = Color(0xFF2A1C15)
    val TextSecondary = Color(0xFF8A6F5C)
    val TextMuted = Color(0xFFBDA491)
    val DeepAccent = Color(0xFFC2410C)
    val Accent = Color(0xFFEA580C)
    val GradStart = Color(0xFFFB923C)
    val GradEnd = Color(0xFFEA580C)
    val Success = Color(0xFF57A075)
    val Warn = Color(0xFFD97706)
    val Avoid = Color(0xFFDC2626)

    // Service icon-tile gradients
    val AartiGrad = listOf(Color(0xFFFCA5A5), Color(0xFFDC2626))
    val ChoghadiyaGrad = listOf(Color(0xFFFB923C), Color(0xFFEA580C))
    val FestivalGrad = listOf(Color(0xFFA5B4FC), Color(0xFF6366F1))
    val PanchangGrad = listOf(Color(0xFFFCD34D), Color(0xFFD97706))
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ExploreScreen(
    onOpenProfile: () -> Unit,
    onOpenDivineImage: () -> Unit,
    onOpenAartis: () -> Unit,
    onOpenChoghadiya: () -> Unit,
    onOpenFestivals: () -> Unit,
    onOpenPanchang: () -> Unit
) {
    Scaffold(
        contentWindowInsets = androidx.compose.foundation.layout.WindowInsets(0),
        containerColor = MaterialTheme.colorScheme.background,
        topBar = {
            AppTopBar(
                title = "Explore",
                leftContent = {
                    IconButton(onClick = onOpenProfile) {
                        Icon(
                            imageVector = Icons.Outlined.Person,
                            contentDescription = "Profile"
                        )
                    }
                }
            )
        }
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .verticalScroll(rememberScrollState())
                .padding(horizontal = 18.dp)
                .padding(top = 8.dp, bottom = BhaktiBottomNavBarDefaults.overlayClearance + 12.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            FeaturedDivineImageCard(onClick = onOpenDivineImage)

            Text("Services", fontSize = 18.sp, fontWeight = FontWeight.ExtraBold, color = ExplorePalette.TextPrimary)

            ServiceRow(
                icon = Icons.Filled.MusicNote,
                gradient = ExplorePalette.AartiGrad,
                title = "Aartis",
                subtitle = "30+ devotional songs with lyrics & audio",
                onClick = onOpenAartis
            )
            ServiceRow(
                icon = Icons.Filled.WbSunny,
                gradient = ExplorePalette.ChoghadiyaGrad,
                title = "Choghadiya",
                subtitle = "Today's auspicious timings for your city",
                onClick = onOpenChoghadiya
            )
            ServiceRow(
                icon = Icons.Filled.Celebration,
                gradient = ExplorePalette.FestivalGrad,
                title = "Festivals",
                subtitle = "Upcoming Hindu festivals & vrat",
                onClick = onOpenFestivals
            )
            ServiceRow(
                icon = Icons.Filled.CalendarMonth,
                gradient = ExplorePalette.PanchangGrad,
                title = "Panchang",
                subtitle = "Tithi, nakshatra, sunrise & sunset today",
                onClick = onOpenPanchang
            )

            Text("Coming soon", fontSize = 18.sp, fontWeight = FontWeight.ExtraBold, color = ExplorePalette.TextPrimary)
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                ComingSoonCard(Icons.Filled.Stars, "Rashifal", "Daily horoscope", Modifier.weight(1f))
                ComingSoonCard(Icons.Filled.GridView, "Kundli", "Birth chart", Modifier.weight(1f))
            }
        }
    }
}

@Composable
private fun FeaturedDivineImageCard(onClick: () -> Unit) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(190.dp)
            .clip(RoundedCornerShape(24.dp))
            .clickable(onClick = onClick)
    ) {
        Image(
            painter = painterResource(id = R.drawable.photo_with_god),
            contentDescription = "Divine Image",
            modifier = Modifier.fillMaxSize(),
            contentScale = ContentScale.Crop
        )
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(
                    Brush.verticalGradient(
                        0.25f to Color.Transparent,
                        1f to Color.Black.copy(alpha = 0.82f)
                    )
                )
        )
        Column(
            modifier = Modifier
                .align(Alignment.BottomStart)
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(6.dp)
        ) {
            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                Icon(Icons.Filled.AutoAwesome, null, tint = Color(0xFFFFE7C6), modifier = Modifier.size(13.dp))
                Text("FEATURED", fontSize = 11.sp, fontWeight = FontWeight.Bold, color = Color(0xFFFFE7C6), letterSpacing = 1.sp)
            }
            Text("Divine Image", fontSize = 22.sp, fontWeight = FontWeight.ExtraBold, color = Color.White)
            Text("Turn your photo into a sacred darshan", fontSize = 13.sp, color = Color.White.copy(alpha = 0.9f))
            Spacer(Modifier.height(2.dp))
            Row(
                modifier = Modifier
                    .clip(RoundedCornerShape(999.dp))
                    .background(Color.White)
                    .padding(horizontal = 14.dp, vertical = 8.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                Text("Create yours", fontSize = 13.sp, fontWeight = FontWeight.Bold, color = ExplorePalette.DeepAccent)
                Icon(Icons.Filled.ChevronRight, null, tint = ExplorePalette.DeepAccent, modifier = Modifier.size(16.dp))
            }
        }
    }
}

@Composable
private fun ServiceRow(
    icon: ImageVector,
    gradient: List<Color>,
    title: String,
    subtitle: String,
    onClick: () -> Unit
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(18.dp))
            .background(ExplorePalette.Card)
            .border(1.dp, ExplorePalette.CardBorder, RoundedCornerShape(18.dp))
            .clickable(onClick = onClick)
            .padding(14.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Box(
            modifier = Modifier
                .size(46.dp)
                .clip(RoundedCornerShape(14.dp))
                .background(Brush.linearGradient(gradient)),
            contentAlignment = Alignment.Center
        ) {
            Icon(icon, null, tint = Color.White, modifier = Modifier.size(22.dp))
        }
        Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(title, fontSize = 15.sp, fontWeight = FontWeight.Bold, color = ExplorePalette.TextPrimary)
            Text(subtitle, fontSize = 12.5.sp, color = ExplorePalette.TextSecondary)
        }
        Icon(Icons.Filled.ChevronRight, null, tint = ExplorePalette.TextMuted, modifier = Modifier.size(22.dp))
    }
}

@Composable
private fun ComingSoonCard(icon: ImageVector, title: String, subtitle: String, modifier: Modifier = Modifier) {
    Column(
        modifier = modifier
            .clip(RoundedCornerShape(18.dp))
            .background(ExplorePalette.Card.copy(alpha = 0.6f))
            .border(1.dp, ExplorePalette.CardBorder, RoundedCornerShape(18.dp))
            .padding(14.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp)
    ) {
        Icon(icon, null, tint = ExplorePalette.TextMuted, modifier = Modifier.size(24.dp))
        Text(title, fontSize = 14.5.sp, fontWeight = FontWeight.Bold, color = ExplorePalette.TextPrimary.copy(alpha = 0.7f))
        Text(subtitle, fontSize = 12.sp, color = ExplorePalette.TextMuted)
        Box(
            modifier = Modifier
                .clip(RoundedCornerShape(999.dp))
                .background(ExplorePalette.TextMuted.copy(alpha = 0.15f))
                .padding(horizontal = 8.dp, vertical = 3.dp)
        ) {
            Text("Coming soon", fontSize = 10.sp, fontWeight = FontWeight.Bold, color = ExplorePalette.TextSecondary)
        }
    }
}

File: BhaktiChatiOS/Sources/Features/Explore/ExploreScreen.swift
import SwiftUI

/// The "Explore" tab: featured Divine Image, service rows (Aartis/Choghadiya/Festivals/
/// Panchang), and cosmetic "Coming soon" placeholders. Mirrors Android's `ExploreScreen`
/// in `ui/screens/explore/ExploreScreen.kt` (design_handoff_bhaktichat_ia).
private enum ExploreRoute: Hashable {
    case divineHome
    case divineCreate(templateId: String)
    case divineResult(creationId: String)
}

struct ExploreScreen: View {
    @EnvironmentObject private var appState: AppState
    @State private var path: [ExploreRoute] = []

    var body: some View {
        NavigationStack(path: $path) {
            ScrollView {
                VStack(alignment: .leading, spacing: 16) {
                    Button {
                        path.append(.divineHome)
                    } label: {
                        FeaturedDivineImageCard()
                    }
                    .buttonStyle(.plain)

                    Text("Services")
                        .font(.system(size: 18, weight: .heavy))
                        .foregroundStyle(ExplorePalette.textPrimary)

                    NavigationLink {
                        AartisScreen()
                    } label: {
                        ServiceRow(
                            systemImage: "music.note",
                            gradient: ExplorePalette.aartiGradient,
                            title: "Aartis",
                            subtitle: "30+ devotional songs with lyrics & audio"
                        )
                    }
                    .buttonStyle(.plain)

                    NavigationLink {
                        ChoghadiyaScreen()
                    } label: {
                        ServiceRow(
                            systemImage: "sun.max.fill",
                            gradient: ExplorePalette.choghadiyaGradient,
                            title: "Choghadiya",
                            subtitle: "Today's auspicious timings for your city"
                        )
                    }
                    .buttonStyle(.plain)

                    NavigationLink {
                        FestivalsScreen()
                    } label: {
                        ServiceRow(
                            systemImage: "party.popper.fill",
                            gradient: ExplorePalette.festivalGradient,
                            title: "Festivals",
                            subtitle: "Upcoming Hindu festivals & vrat"
                        )
                    }
                    .buttonStyle(.plain)

                    NavigationLink {
                        PanchangScreen()
                    } label: {
                        ServiceRow(
                            systemImage: "calendar",
                            gradient: ExplorePalette.panchangGradient,
                            title: "Panchang",
                            subtitle: "Tithi, nakshatra, sunrise & sunset today"
                        )
                    }
                    .buttonStyle(.plain)

                    NavigationLink {
                        WallpapersScreen()
                    } label: {
                        ServiceRow(
                            systemImage: "photo.on.rectangle.angled",
                            gradient: ExplorePalette.wallpaperGradient,
                            title: "Wallpapers",
                            subtitle: "Deity wallpapers to save & share"
                        )
                    }
                    .buttonStyle(.plain)

                    Text("Coming soon")
                        .font(.system(size: 18, weight: .heavy))
                        .foregroundStyle(ExplorePalette.textPrimary)

                    HStack(spacing: 12) {
                        ComingSoonCard(systemImage: "sparkles", title: "Rashifal", subtitle: "Daily horoscope")
                        ComingSoonCard(systemImage: "square.grid.2x2", title: "Kundli", subtitle: "Birth chart")
                    }
                }
                .padding(.horizontal, 16)
                .padding(.top, 8)
                .padding(.bottom, BhaktiBottomNavBar.overlayClearance + 12)
            }
            .bhaktiPageBackground()
            .bhaktiHideNavigationBar()
            .safeAreaInset(edge: .top, spacing: 0) { topBar }
            .navigationDestination(for: ExploreRoute.self) { route in
                switch route {
                case .divineHome:
                    DivineImageHomeScreen { templateId in
                        path.append(.divineCreate(templateId: templateId))
                    }
                case let .divineCreate(templateId):
                    if let template = DivineTemplateCatalog.byId(templateId) {
                        DivineImageCreateScreen(template: template) { creationId in
                            path.append(.divineResult(creationId: creationId))
                        }
                        .environmentObject(appState)
                    } else {
                        Text("Template not found")
                            .foregroundStyle(BhaktiTheme.textSecondary)
                            .bhaktiPageBackground()
                    }
                case let .divineResult(creationId):
                    DivineImageResultScreen(creationId: creationId)
                }
            }
        }
    }

    // MARK: - Top bar

    private var topBar: some View {
        AppTopBar(
            leftContent: {
                NavigationLink {
                    ProfileScreen()
                } label: {
                    Image(systemName: "person.circle")
                        .font(.system(size: 24, weight: .medium))
                        .foregroundStyle(BhaktiTheme.textPrimary)
                }
                .buttonStyle(.plain)
                .accessibilityLabel("Profile")
            },
            centerContent: {
                Text("Explore")
                    .font(.system(size: 17, weight: .semibold))
                    .foregroundStyle(BhaktiTheme.textPrimary)
            },
            rightContent: {
                EmptyView()
            }
        )
        .padding(.horizontal, 16)
        .background(BhaktiTheme.background.opacity(0.96))
        .overlay(alignment: .bottom) {
            Rectangle()
                .fill(BhaktiTheme.border.opacity(0.6))
                .frame(height: 0.5)
        }
    }
}

private struct FeaturedDivineImageCard: View {
    var body: some View {
        ZStack(alignment: .bottomLeading) {
            PackageAssetLoader.image(named: "photo_with_god")
                .resizable()
                .scaledToFill()

            LinearGradient(
                stops: [
                    .init(color: .clear, location: 0.25),
                    .init(color: Color.black.opacity(0.82), location: 1)
                ],
                startPoint: .top,
                endPoint: .bottom
            )

            VStack(alignment: .leading, spacing: 6) {
                HStack(spacing: 4) {
                    Image(systemName: "sparkles")
                        .font(.system(size: 13, weight: .bold))
                        .foregroundStyle(Color(red: 0xFF / 255, green: 0xE7 / 255, blue: 0xC6 / 255))
                    Text("FEATURED")
                        .font(.system(size: 11, weight: .bold))
                        .tracking(1)
                        .foregroundStyle(Color(red: 0xFF / 255, green: 0xE7 / 255, blue: 0xC6 / 255))
                }
                Text("Divine Image")
                    .font(.system(size: 22, weight: .heavy))
                    .foregroundStyle(.white)
                Text("Turn your photo into a sacred darshan")
                    .font(.system(size: 13))
                    .foregroundStyle(Color.white.opacity(0.9))
                Spacer().frame(height: 2)
                HStack(spacing: 4) {
                    Text("Create yours")
                        .font(.system(size: 13, weight: .bold))
                        .foregroundStyle(ExplorePalette.deepAccent)
                    Image(systemName: "chevron.right")
                        .font(.system(size: 12, weight: .bold))
                        .foregroundStyle(ExplorePalette.deepAccent)
                }
                .padding(.horizontal, 14)
                .padding(.vertical, 8)
                .background(Color.white)
                .clipShape(Capsule())
            }
            .padding(16)
        }
        // Android uses a fixed 190dp here with the same padding/spacing/font sizes, and that
        // exact number was ported verbatim — but SF Pro renders this text stack a bit taller
        // than Android's Roboto at the same point sizes, which was clipping the "Create
        // yours" pill against the card's rounded bottom edge. minHeight keeps the same
        // visual size in the common case while letting it grow if content needs more.
        .frame(minHeight: 190)
        .clipShape(RoundedRectangle(cornerRadius: 24))
    }
}

private struct ServiceRow: View {
    let systemImage: String
    let gradient: [Color]
    let title: String
    let subtitle: String

    var body: some View {
        HStack(spacing: 12) {
            RoundedRectangle(cornerRadius: 14)
                .fill(LinearGradient(colors: gradient, startPoint: .topLeading, endPoint: .bottomTrailing))
                .frame(width: 46, height: 46)
                .overlay(
                    Image(systemName: systemImage)
                        .font(.system(size: 19, weight: .semibold))
                        .foregroundStyle(.white)
                )

            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(.system(size: 15, weight: .bold))
                    .foregroundStyle(ExplorePalette.textPrimary)
                Text(subtitle)
                    .font(.system(size: 12.5))
                    .foregroundStyle(ExplorePalette.textSecondary)
            }

            Spacer()

            Image(systemName: "chevron.right")
                .font(.system(size: 18, weight: .semibold))
                .foregroundStyle(ExplorePalette.textMuted)
        }
        .padding(14)
        .background(ExplorePalette.card)
        .overlay(
            RoundedRectangle(cornerRadius: 18).stroke(ExplorePalette.cardBorder, lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: 18))
    }
}

private struct ComingSoonCard: View {
    let systemImage: String
    let title: String
    let subtitle: String

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Image(systemName: systemImage)
                .font(.system(size: 20, weight: .semibold))
                .foregroundStyle(ExplorePalette.textMuted)
            Text(title)
                .font(.system(size: 14.5, weight: .bold))
                .foregroundStyle(ExplorePalette.textPrimary.opacity(0.7))
            Text(subtitle)
                .font(.system(size: 12))
                .foregroundStyle(ExplorePalette.textMuted)
            Text("Coming soon")
                .font(.system(size: 10, weight: .bold))
                .foregroundStyle(ExplorePalette.textSecondary)
                .padding(.horizontal, 8)
                .padding(.vertical, 3)
                .background(ExplorePalette.textMuted.opacity(0.15))
                .clipShape(Capsule())
        }
        .padding(14)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(ExplorePalette.card.opacity(0.6))
        .overlay(
            RoundedRectangle(cornerRadius: 18).stroke(ExplorePalette.cardBorder, lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: 18))
    }
}
